tests/phpunit/unit-tests/class-llms-test-mime-type-extractor.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Test LLMS_Mime_Type_Extractor
+ *
+ * @package LifterLMS/Tests
+ *
+ * @group mime_type_extractor
+ *
+ * @since [version]
+ * @version [version]
+ */
+class LLMS_Test_Mime_Type_Extractor extends LLMS_UnitTestCase {
+
+	/**
+	 * Test files.
+	 *
+	 * @var array
+	 */
+	protected $files = array(
+		'json' => 'import-error.json',
+		'jpg'  => 'christian-fregnan-unsplash.jpg',
+	);
+
+	/**
+	 * Test from_file_path() for a file with a mime-type that exists
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_mime_type_in_list() {
+
+		global $lifterlms_tests;
+		$this->assertEquals(
+			'image/jpeg',
+			LLMS_Mime_Type_Extractor::from_file_path( $lifterlms_tests->assets_dir . $this->files['jpg'] )
+		);
+
+	}
+
+	/**
+	 * Test from_file_path() for a mime-type not found in our list
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_mime_type_not_in_list() {
+
+		global $lifterlms_tests;
+
+		// I expect json to be recognized either by one of the fallback functions as 'text/plain'.
+		if ( function_exists( 'finfo_file' ) ) {
```

Comment:
travis tests are failing on php 7.4 where `finfo_file()` exists.

I've been able to reproduce locally on my end too. It looks like `finfo_file()` on 7.4 at least responds with `application/json` for the tested file but it returns as `text/html` when using the `sample-course.json` file found in the core plugin `sample-data` directory.

A bit of research shows that the function uses the file's "signature" to determine the filetype and JSON files can have varied signatures based on their content. I guess the sample course has some HTML in it causing it to look like an HTML from a "signature" perspective.

I don't fully understand it all but I'm leaning towards simply scrapping this test since it's going to respond differently per PHP version and since for our purposes we won't be using it to identify JSON files and we can simply move on.

Reply:
This is annoying I just wanted to cover the 'not in list'. :/

Reply:
Ahh i see... I'm not sure if there's going to be a way to do that given the way finfo_file works

Maybe creating a text file in the tests/assets directory with no content that has a fake file extension, eg: `fake-mime.fakemime`

The signature on that might still read as a text file (or maybe not) but it should fallback correctly anyway then, right?

Reply:
Also, sorry.

Reply:
I actually tested that locally and when empty (created with `touch`) it was recognized as empty inode, if with text inside, it was recognized as text...

Reply:
@eri-trabiccolo cool you fixed it!

Reply:
well you know, I cheated, I mean we weren't interested in a deep test :D